src/main/java/org/jabref/logic/importer/util/MetaDataParser.java

```diff
@@ -49,68 +52,62 @@ public MetaData parse(MetaData metaData, Map<String, String> data, Character key
         List<String> defaultCiteKeyPattern = new ArrayList<>();
         Map<EntryType, List<String>> nonDefaultCiteKeyPatterns = new HashMap<>();
 
-        for (Map.Entry<String, String> entry : data.entrySet()) {
+        // process groups (GROUPSTREE and GROUPSTREE_LEGACY) at the very end (otherwise it can happen that not all dependent data are set)
+        Stream<Map.Entry<String, String>> entrySetStream = data.entrySet().stream().sorted(groupsLast());
```

Comment:
Just an idea: Can creating streams theoretically be omitted by using `List.copyof(data.entrySet()).sort(groupsLast())` ?